test/compatibility/upgrade_test.lua

```diff
@@ -0,0 +1,209 @@
+        srv:join_cluster(g.cluster.main_server)
+        g.cluster:retrying({}, function() srv:connect_net_box() end)
+    end
+    g.cluster.main_server:setup_replicaset(g.cluster.replicasets[1])
+    g.cluster:bootstrap_vshard()
+    g.cluster:wait_until_healthy()
+
+    t.assert_not_equals(version(leader), 'scm-1')
+    t.assert_not_equals(version(replica), 'scm-1')
+
+    -- Setup data model
+    leader.net_box:eval([[
+        box.schema.space.create('test', {
+            format = {
+                {name = 'bucket_id', type = 'unsigned', is_nullable = false},
+                {name = 'record_id', type = 'unsigned', is_nullable = false},
+                {name = 'alias', type = 'string', is_nullable = false},
+            }
+        })
+
+        box.space.test:create_index('primary', {
+            unique = true,
+            parts = {{'record_id', 'unsigned', is_nullable = false}},
+        })
+        box.space.test:create_index('bucket_id', {
+            unique = false,
+            parts = {{'bucket_id', 'unsigned', is_nullable = false}},
+        })
+    ]])
+
+    -- Start streaming
+    local reference = {}
+    local function _insert(cnt)
+        local ret, err = g.cluster.main_server.net_box:eval([[
+            return package.loaded.vshard.router.callrw(
+                1, 'box.space.test:insert', {{1, ...}}
+            )
+        ]], {cnt, g.cluster.main_server.alias})
+
+        if ret == nil then
+            log.error('CNT %d: %s', cnt, yaml.encode(err))
+        else
+            table.insert(reference, ret)
+        end
+        return true
+    end
+    local highload_cnt = 0
+
+
+    local highload_fiber = fiber.new(function()
+        log.warn('Highload started >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>')
+        while true do
+            highload_cnt = highload_cnt + 1
+            local ok, err = errors.pcall('E', _insert, highload_cnt)
+            if ok == nil then
+                log.error('CNT %d: %s', highload_cnt, err)
+            end
+            fiber.sleep(0.01)
+        end
+    end)
+    highload_fiber:name('test.highload')
+    fiber.sleep(0.2)
+
+    -- Upgrade replica
+    replica:stop()
+    replica.chdir = nil
+    replica:start()
+    g.cluster:retrying({}, function() replica:connect_net_box() end)
+    wish_state(replica, 'RolesConfigured')
+
+    t.assert_not_equals(version(leader), 'scm-1')
+    t.assert_equals(version(replica), 'scm-1')
+    t.assert_equals(upstream_info(replica), {status = 'follow'})
+
+    -- Switch the leadership
+    leader.net_box:eval([[
+        local errors = require('errors')
+        local cartridge = require('cartridge')
+        errors.assert('E', cartridge.admin_edit_topology(...))
+    ]], {{
+        replicasets = {{
+            uuid = leader.replicaset_uuid,
+            failover_priority = {replica.instance_uuid},
+        }}
+    }})
+    g.cluster.main_server = replica
+
+    fiber.sleep(0.2)
```

Comment:
why?

Reply:
I've changed it with more clear `g.cluster.retying`. The idea is to wait for workload switching before killing leader